app/api-v1/services/orderService.js

```diff
@@ -0,0 +1,17 @@
+const { postOrderDb } = require('../../db')
+
+async function postProject(reqBody) {
+  // Will add a get function at a later date to check for duplication
+
+  const result = await postOrderDb(reqBody)
+
+  if (!result) {
+    return { statusCode: 400, result: {} }
```

Comment:
on `POST attachment` and `POST recipe` there's a `throw new BadRequestError` you can throw instead of returning the response directly. The error handling middleware then catches the error and responds with the code and message you set in `BadRequestError`

Reply:
so the error response would end up happening here https://github.com/digicatapult/inteli-api/blob/ca869047f74257bc99a18047f9185a619eca4380/app/utils/errors.js#L21

Reply:
it means you shouldn't need `try/catch` anywhere on the route. You can throw your own errors or something like `knex` can throw an error, and it will be caught by the error handling middleware

Reply:
In the code block above, where am I getting the `err` from? should it be `statusCode.code`?

Reply:
the `err` is the whole `BadRequestError` object